### ISSUE TITLE
Pylint wrongly detects alias properties in slotted classes

### ISSUE THREAD

**Issue by mathieui** [PyCQA/pylint#2439]
Hi, we encountered the issue in our CI which runs a pylint -E.
The issue does not happen in non-slotted classes, as far as I am aware.

### Steps to reproduce

Run pylint -E on the following file:

```python
class TestClass:
    __slots__ = ['_i']
    def __init__(self):
        self._i = 0

    @property
    def i(self):
        return self._i

    @i.setter
    def i(self, v):
        self._i = v

    other = i

instance = TestClass()
instance.other = 42
print(instance.i)
```

### Current behavior
```
************* Module test
test.py:17:0: E0237: Assigning to attribute 'other' not defined in class slots (assigning-non-slot)
```
(and other errors in the code using that class)

### Expected behavior

No error

### pylint --version output

```
pylint 2.1.1
astroid 2.0.4
Python 3.7.0 (default, Jul 15 2018, 10:44:58) 
[GCC 8.1.1 20180531]
```
